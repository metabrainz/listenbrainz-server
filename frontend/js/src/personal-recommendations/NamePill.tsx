import * as React from "react";
import { faTimesCircle } from "@fortawesome/free-solid-svg-icons";
import { isFunction } from "lodash";
import ListenControl from "../common/listens/ListenControl";

export type NamePillProps = {
  title: string;
  closeAction?: (event: React.MouseEvent) => void;
};

function NamePill(props: NamePillProps) {
  const { title, closeAction } = props;
  return (
    <div className="pill secondary active">
      <span>{title}</span>
      {isFunction(closeAction) && (
        <ListenControl
          title="Remove"
          text=""
          icon={faTimesCircle}
          action={closeAction}
          isDropdown={false}
        />
      )}
    </div>
  );
}

export default NamePill;
