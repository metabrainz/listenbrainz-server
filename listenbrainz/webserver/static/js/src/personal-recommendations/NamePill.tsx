import * as React from "react";
import { faTimesCircle } from "@fortawesome/free-solid-svg-icons";
import { isFunction } from "lodash";
import ListenControl from "../listens/ListenControl";

export type NamePillProps = {
  title: string;
  closeAction?: (event: React.MouseEvent) => void;
  closeButton?: Boolean;
};

const NamePill = (props: NamePillProps) => {
  const { title, closeAction, closeButton } = props;
  return (
    <div className="pill">
      <div>
        <span>{title}</span>
        {closeButton && (
          <ListenControl text="" icon={faTimesCircle} action={closeAction} />
        )}
      </div>
    </div>
  );
};

export default NamePill;
