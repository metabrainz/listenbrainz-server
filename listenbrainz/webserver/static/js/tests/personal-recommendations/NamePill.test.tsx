import * as React from "react";
import { mount } from "enzyme";

import NamePill, {
  NamePillProps,
} from "../../src/personal-recommendations/NamePill";

// Font Awesome generates a random hash ID for each icon everytime.
// Mocking Math.random() fixes this
// https://github.com/FortAwesome/react-fontawesome/issues/194#issuecomment-627235075
jest.spyOn(global.Math, "random").mockImplementation(() => 0);

const mockAction = jest.fn();

const props: NamePillProps = {
  title: "foobar",
  closeAction: mockAction,
};

describe("NamePill", () => {
  it("renders correctly", () => {
    const wrapper = mount(<NamePill {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it("works when you click", () => {
    const wrapper = mount(<NamePill {...props} />);
    wrapper.find("button").at(0).simulate("click");
    expect(mockAction).toHaveBeenCalledTimes(1);
  });

  it("renders a close button if closeAction is a function", () => {
    const wrapper = mount(<NamePill title="foobar" closeAction={() => {}} />);
    expect(wrapper.find("button").exists()).toBeTruthy();
  });

  it("does not render a close button if closeAction is not a function (undefined)", () => {
    const wrapper = mount(<NamePill title="foobar" />);
    expect(wrapper.find("button").exists()).toBeFalsy();
  });
});
