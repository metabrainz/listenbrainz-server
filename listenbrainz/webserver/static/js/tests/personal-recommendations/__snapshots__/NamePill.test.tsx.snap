// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NamePill renders correctly 1`] = `
<NamePill
  closeAction={[MockFunction]}
  closeButton={true}
  title="foobar"
>
  <div
    className="pill"
  >
    <div>
      <span>
        foobar
      </span>
      <ListenControl
        action={[MockFunction]}
        icon={
          Object {
            "icon": Array [
              512,
              512,
              Array [],
              "f057",
              "M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm121.6 313.1c4.7 4.7 4.7 12.3 0 17L338 377.6c-4.7 4.7-12.3 4.7-17 0L256 312l-65.1 65.6c-4.7 4.7-12.3 4.7-17 0L134.4 338c-4.7-4.7-4.7-12.3 0-17l65.6-65-65.6-65.1c-4.7-4.7-4.7-12.3 0-17l39.6-39.6c4.7-4.7 12.3-4.7 17 0l65 65.7 65.1-65.6c4.7-4.7 12.3-4.7 17 0l39.6 39.6c4.7 4.7 4.7 12.3 0 17L312 256l65.6 65.1z",
            ],
            "iconName": "times-circle",
            "prefix": "fas",
          }
        }
        text=""
      >
        <button
          aria-label=""
          disabled={false}
          onClick={[MockFunction]}
          title=""
          type="button"
        >
          <FontAwesomeIcon
            border={false}
            className=""
            fixedWidth={false}
            flip={null}
            icon={
              Object {
                "icon": Array [
                  512,
                  512,
                  Array [],
                  "f057",
                  "M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm121.6 313.1c4.7 4.7 4.7 12.3 0 17L338 377.6c-4.7 4.7-12.3 4.7-17 0L256 312l-65.1 65.6c-4.7 4.7-12.3 4.7-17 0L134.4 338c-4.7-4.7-4.7-12.3 0-17l65.6-65-65.6-65.1c-4.7-4.7-4.7-12.3 0-17l39.6-39.6c4.7-4.7 12.3-4.7 17 0l65 65.7 65.1-65.6c4.7-4.7 12.3-4.7 17 0l39.6 39.6c4.7 4.7 4.7 12.3 0 17L312 256l65.6 65.1z",
                ],
                "iconName": "times-circle",
                "prefix": "fas",
              }
            }
            inverse={false}
            listItem={false}
            mask={null}
            pull={null}
            pulse={false}
            rotation={null}
            size={null}
            spin={false}
            swapOpacity={false}
            symbol={false}
            title=""
            transform={null}
          >
            <svg
              aria-hidden="true"
              className="svg-inline--fa fa-times-circle fa-w-16 "
              data-icon="times-circle"
              data-prefix="fas"
              focusable="false"
              role="img"
              style={Object {}}
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm121.6 313.1c4.7 4.7 4.7 12.3 0 17L338 377.6c-4.7 4.7-12.3 4.7-17 0L256 312l-65.1 65.6c-4.7 4.7-12.3 4.7-17 0L134.4 338c-4.7-4.7-4.7-12.3 0-17l65.6-65-65.6-65.1c-4.7-4.7-4.7-12.3 0-17l39.6-39.6c4.7-4.7 12.3-4.7 17 0l65 65.7 65.1-65.6c4.7-4.7 12.3-4.7 17 0l39.6 39.6c4.7 4.7 4.7 12.3 0 17L312 256l65.6 65.1z"
                fill="currentColor"
                style={Object {}}
              />
            </svg>
          </FontAwesomeIcon>
           
        </button>
      </ListenControl>
    </div>
  </div>
</NamePill>
`;
